### ISSUE TITLE
Song Requests: Youtube songs keep restarting every few seconds

### ISSUE THREAD

**Issue by TyrenDe** [SaviorXTanren/mixer-mixitup#295]
Okay, I root caused this to a strange combinations. To repro.  Enable Overlays, Enable YouTube Song requests, create a command to Enable/Disable the song request queue, create a command to queue a song.  Then run the toggle command 3-6 times. Now, queue a song.

The problem is, the action that disables/enables the queue triggers the setting to change, and this gets refreshed in the RefreshRequestsList method.  And then the check box is toggled, and that triggers the events EnableSongRequestsToggleButton_Checked and EnableSongRequestsToggleButton_Unchecked, which confuses the system as the initialize gets called multiple times.  And at the end of the initialize, it creates a new TokenSource, which makes it impossible to stop the previously started background thread.

Long story short: Multiple threads are now running the song request background thread, which causes all sorts of confusion.

**Comment by SaviorXTanren** [comment]
Fixed via https://github.com/SaviorXTanren/mixer-mixitup/pull/296
